[Bug 16085] build: Documentation for building with Visual Studio 2010 Express

peter-b opened the pull request:
This pull request fixes some issues with Windows native builds, and provides the long-missing `INSTALL-win.md` documentation.

The main (possibly controversial) change here is to require cpptest test source files to be listed explicitly in the gyp files, rather than automatically scanning for them.

peter-b committed:
build: Use %ProgramFiles(x86)% environment variable during build.

When compiling on newer versions of Windows, Visual Studio may be located
in either `C:\Program Files` or `C:\Program Files (x86)`, depending on
whether the operating system is 32-bit or 64-bit.

peter-b committed:
build: Add Perl script to help decode the version file.

The build configuration needs to parse the `version` file to generate
some version-dependent parameters.  Unfortunately passing literal
regular expressions on the perl command line was difficult to escape
properly on Windows.  This moves the parsing into a helper script.

peter-b committed:
build: User windows.h rather than afxres.h

afxres.h is not included in Visual Studio 2010 Express

peter-b committed:
build: Explicitly list cpptest source files.

This is because I couldn't find a completely portable command to list files
matching a specific pattern in a subdirectory (or even a decent substitute on
Windows for `ls -1 path/to/*.cpp`).

This isn't a massive loss because we already explicitly list all of our other
C++ source files.

peter-b committed:
[Bug 16085] docs: Add INSTALL-win.md file.

Add some basic instructions for compiling LiveCode on Windows.

livecodefraser commented:
I am confused. GitHub tells me that the final commit adds 0 new files and lists no changes... definitely doesn't show any new .md files, for whatever reason.

peter-b commented:
@livecodefraser Weird.  The final commit looks okay to me. :smiling_imp:

livecodefraser commented:
Hehe, no change in the commit hash. Honest. :angel:

@livecode-vulcan review okay 2b597cd

livecode-vulcan commented:
💙 review by @livecodefraser ok 2b597cd9c3dd40faf03724eea6c80e6e99471c82

livecode-vulcan commented:
😞 test failure 2b597cd9c3dd40faf03724eea6c80e6e99471c82
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/878)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/892)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/879)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/882)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/892)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/902)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/895)

peter-b commented:
@livecode-vulcan retry 2b597cd

livecode-vulcan commented:
😞 test failure 2b597cd9c3dd40faf03724eea6c80e6e99471c82
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/880)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/894)
- try-community-linux-x86_64: [pending](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/881)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/884)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/894)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/904)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/897)

peter-b commented:
@livecode-vulcan retry 2b597cd

livecode-vulcan commented:
😞 test failure 2b597cd9c3dd40faf03724eea6c80e6e99471c82
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/881)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/895)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/882)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/885)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/895)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/905)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/898)

peter-b committed:
build: Look for 64-bit ActivePerl during Win builds

The make.cmd script for Windows added `C:\Perl` to the search path,
but omitted `C:\Per64`, which is where the x64 version of
ActiveState Perl gets installed by default.

livecodefraser commented:
@livecode-vulcan review okay f85a686

livecode-vulcan commented:
💙 review by @livecodefraser ok f85a6860180468ec4c9cbe4c3d73a4037ad85c9f

livecode-vulcan commented:
😞 test failure f85a6860180468ec4c9cbe4c3d73a4037ad85c9f
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/883)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/897)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/884)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/887)
- try-community-ios: [pending](https://vulcan.livecode.com/builders/try-community-ios/builds/897)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/907)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/900)

peter-b committed:
build: Deal with missing %ProgramFiles(x86)% in Wine

When compiling on Windows x64, the PATH must be constructed using
the 32-bit version of the "Program Files" directory.  Normally,
this can be located using the `%ProgramFiles(x86)%` environment
variable.  Unfortunately, Wine's CMD implementation really doesn't
like environment variables with parentheses in.  Try and guess a
suitable directory be another means.

livecodefraser commented:
@livecode-vulcan review okay 9fa8e45

livecode-vulcan commented:
💙 review by @livecodefraser ok 9fa8e45344a57a0a9f258cf143fd87028c581d0e

livecode-vulcan commented:
😞 test failure 9fa8e45344a57a0a9f258cf143fd87028c581d0e
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/885)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/898)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/885)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/888)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/898)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/908)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/901)

peter-b commented:
(╯°□°)╯︵ ┻━┻

livecodefraser commented:
@livecode-vulcan review okay 4196d18

livecode-vulcan commented:
💙 review by @livecodefraser ok 4196d18df5ecd7e740859fd5a8dc0e1ba97b7fb7

livecode-vulcan commented:
😎 test success 4196d18df5ecd7e740859fd5a8dc0e1ba97b7fb7
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/895)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/905)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/894)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/896)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/906)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/916)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/909)

livecodefraser commented:
@montegoulding I've not tried with VS2013 but I'd be interested to hear how it goes. We've been thinking of moving to it as (I believe) it still supports targeting XP even though it won't run on it.

montegoulding commented:
OK, will let you know

montegoulding commented:
@livecodefraser Fails miserably... not sure if you want build output
